### PR TITLE
Feature saadc

### PR DESCRIPTION
**Pull request by ajtwlsdlqp**
- Add I2S in peripheral_alloc.
- Add NRF52Pin saadc

### CODE REVIEW

**Comment by finneyj** [comment]
Awesome - thanks @ajtwlsdlqp!

I'm working on a DMA streaming branch for the SAADC too, but I haven't implemented these interfaces yet...

Rather than two independent implementations, would it be ok if I merge your contribution into that, so we can have a unified implementation?

File: inc/NRF52Pin.h [hunk 1]
```diff
@@ -36,6 +36,7 @@ DEALINGS IN THE SOFTWARE.
 #define ID_NRF52_PIN_LO  (DEVICE_ID_IO_P0 + 36)
 
 #define NRF52PIN_PWM_CHANNEL_MAP_SIZE        4
+#define NRF52PIN_SSADC_MHANNEL_MAP_SIZE      8
```
Comment (mmoskal) [inline inc/NRF52Pin.h:39]: Typos? `NRF52PIN_SAADC_CHANNEL_MAP_SIZE`

File: source/NRF52Pin.cpp [hunk 2]
```diff
@@ -379,25 +411,54 @@ int NRF52Pin::setServoValue(int value, int range, int center)
...
+    }
+
+    // no existing channel found
+    if (channel == -1) return DEVICE_NOT_SUPPORTED;
+    else
+    {
+        initialiseSAADC();
+    }
+    
+    // Configure SAADC singled-ended channel, Internal reference (0.6V) and 1/6 gain.
```
Comment (mmoskal) [inline source/NRF52Pin.cpp:438]: if this comment correct? it looks like 1/4 gain and 1/4 voltage?

Comment (ajtwlsdlqp) [reply source/NRF52Pin.cpp:438]: Sorry, i will do recheck n fix it

File: inc/NRF52Pin.h (hunk 1 above)
Comment (ajtwlsdlqp) [reply inc/NRF52Pin.h:39]: The nRF52 SAADC has 0 to 7 channels, which is used to specify and search channels.

Comment (mmoskal) [reply inc/NRF52Pin.h:39]: I'm just saying you have `SSADC` instead of `SAADC` and `MHANNEL` instead of `CHANNEL` in the name ;)

Comment (ajtwlsdlqp) [reply inc/NRF52Pin.h:39]: oops, i will recheck that part also :) Thanks for review

**Comment by microbit-carlos** [comment]
Thank you for the PR @ajtwlsdlqp! We've implemented ADC in parallel and I believe this feature is already present in the code base, so this change is no longer required. Sorry we didn't get a chance to merge this around that time, and thanks for your time and effort!

**Pull request closed without merging**